File: 5 - Ohjelmoinnin syventavat tekniikat/Viikko 1/2 - Java/Opiskelija.java
public class Opiskelija {
    // Tietojäsenet alkuun
        private String mNimi;
        private String mOpiskelijaNumero;
        private int mOpintopisteet;

        public Opiskelija(){
            mNimi = "Tuntematon";
            mOpiskelijaNumero = "N/A";
            mOpintopisteet = 0;
        }

        public Opiskelija(String aNimi, String aOpiskelijaNumero){
            mNimi = aNimi;
            mOpiskelijaNumero = aOpiskelijaNumero;
        }

        public void setNimi(String aNimi){
            mNimi = aNimi;
        }

        public void setOpiskelijaNumero(String aOpiskelijaNumero){
            mOpiskelijaNumero = aOpiskelijaNumero;
        }

        public void setOpintopisteet(int aOpintopisteet){
            mOpintopisteet = aOpintopisteet;
        }

        public void tulostaTiedot(){
            System.out.println("Nimi: " + mNimi);
            System.out.println("Opiskelija numero: " + mOpiskelijaNumero);
            System.out.println("Opintopisteet: " + mOpintopisteet);
        }
}

File: 6 - Ohjelmoinnin syventavat tekniikat/C++/Viikko 1/Opiskelija.cpp
#include "Opiskelija.h"

Opiskelija::Opiskelija() : mNimi("Tuntematon"), mOpiskelijanumero("00000000"), mOpintopisteet(0){
    cout << "Opiskelijan oletusrakentaja" << endl;
}
Opiskelija::Opiskelija(string aNimi, string aOpiskelijanumero) : mNimi(aNimi), mOpiskelijanumero(aOpiskelijanumero), mOpintopisteet(0){
    cout << "Opiskelijan 2 parametrinen rakentaja"<< endl;
}
Opiskelija::~Opiskelija(){
    cout << "Opiskelija " << mNimi << " vapautettu muistista "<< endl;
}
void Opiskelija::setNimi(string aNimi) {
   mNimi = aNimi;
}
void Opiskelija::setOpiskelijanumero(string aOpiskelijanumero) {
    mOpiskelijanumero = aOpiskelijanumero;
}
void Opiskelija::setOpintopisteet(int aOpintopisteet) {
    mOpintopisteet = aOpintopisteet;
}
void Opiskelija::tulostaTiedot() {
    cout << "Nimi: " << mNimi << endl;
    cout << "Opiskelijanumero: " << mOpiskelijanumero << endl;
    cout << "Opintopisteet: " << mOpintopisteet << endl;
}


